### DISCUSSION TITLE
Cannot log in

### DISCUSSION THREAD

**Discussion by Enquelare** [hultenvp/solis-sensor#244] [General]
I'm not sure but it seems impossible for me to login. 

Any suggestions on login when stating:
"Cannot login with provided URL and credentials"

Just have the API activated, checked for both url's 
https://www.soliscloud.com:13333/
https://www.soliscloud.com:13333* 

tried username and mailadress with the provided keys/secrets and station.

**Comment by stguz** [comment]
Hi, 
i have the same Problem and Report "Cannot login with provided URL and credentials.

**Comment by sjoram** [comment]
Hi all - I have never commented here but was following the thread for a good while! If it is of any use and just to add my experience.....
I too had the "Cannot login with provided URL and credentials" error last week. After the guys fixed their end I submitted a ticket to Solis. They came back to me quite quickly asking for my key and secret. I sent those yesterday and today I got a response from them saying the issue was resolved. I was able to connect today and all the fields are populating! It does indeed appear to be on a per user basis :(

**Comment by sjoram** [comment]
On a separate note and apologies if asked elsewhere - I don't suppose the API / channel can be used to 'write' back to the inverter? I have 'Inverter Control' enabled so can write using the App but would be awesome if I could use this method...... I live in hope.......

**Comment by stguz** [comment]
I send a ticket today. I will see in the next days.
Thx for Info.

**Comment by ghost** [comment]
I had the same issue.
When I treid without the trailing slash, it worked.
https://www.soliscloud.com:13333

**Reply by hultenvp** [reply]
That's odd, that's no longer necessary since 3.2.0

**Reply by ghost** [reply]
I tried multiple times earlier where I copied the values from Solid Cloud, hence the trailing slash, without success.
When I came back a some hours later I did the same, but at that point I noticed that the UI didn't have the trailing slash and removed that from the Solis information and it worked.
So it's either the trailing slash or I had to wait a while for my API info at Solid to become active.

**Reply by ghost** [reply]
According to Solid the API key is activated without delay.

**Reply by GIMickey** [reply]
Taking away the / at the end of the URL also worked for me, thanks

**Reply by abhijithshaji** [reply]
this worked. ty

**Comment by jseppeli** [comment]
I can't log in
Tried with ginlong_v2 and entered username, password and plant ID but it fails with that the screen says "auth"
As plant ID I have tried both the short and long plant ID

**Reply by rogerwilcou** [reply]
Hi, i have the same problem, all the way it works fine. But when i need to enter my information (username, keyID, keysecret and plant number of username password and plant number) it went wrong. I tried ginlong_v2 and soliscloud. Anyone encountered the same problem and found a solution?
![solis](https://github.com/hultenvp/solis-sensor/assets/156460892/86877158-57cf-459a-8ec6-2af71537945b)

**Comment by Tudesco** [comment]
I can’t log in. Solis support asked to share the API key ID & Secret key. Final answer was “The Home assistant Software is not working well with the Solis cloud API. Please use POSTMAN app for that".

**Reply by juant07** [reply]
try taking away the / at the end of the URL worked for me

**Reply by Tudesco** [reply]
Yes, I did without results. Also to dismiss de couple of zero already writen in the Secret key field. I don't understand why Ginlong doesn't clear both faults once everybody knows are wrong. Thank you, anyway, for the suggestion. In fact in the field PV portal is written without / but when they send all de parameters to be inserted they write the / .

**Reply by hultenvp** [reply]
Trailing slashes are already ignored since months. Cannot be the problem nor solution.
This integration is working as well as their API is. So far, most problems have been server side. So them suggesting to use Postman is just for their convenience.
Try the test app as described [here](https://github.com/hultenvp/solis-sensor/wiki/Soliscloud-issues#testing-issues).
See what errors you get. If it's the "abnormal data" issue then it's a soliscloud issue and you should raise a ticket with them.

**Reply by The-Fincher** [reply]
I removed the trailing / and it worked fine

**Reply by Tudesco** [reply]
Launched test app this is the result

_

> C:\Users\YYYY\Downloads\solis-sensor-master\solis-sensor-master\test>C:\Users\YYyy\AppData\Local\Programs\Python\Python310\python.exe apitest_async.py -v
> Verbose
> Using predefined station list
> station_detail(station_id= xxxxxxxxx ): Failed with: Http status code: 403
> collector_list( xxxxxxxxx ): Failed with: Http status code: 403
> inverter_list(station_id = xxxxxxxxx ): Failed with: Timeout error occurred
> station_day(station_id = xxxxxxxxx ): Failed with: Timeout error occurred
> station_month(station_id = xxxxxxxxx ): Failed with: API returned an error: 0, message='Attempt to decode JSON with unexpected mimetype: text/html', url=URL('https://www.soliscloud.com:13333/v1/api/stationMonth'), error code: None, response None
> station_year(station_id = xxxxxxxxx ): Failed with: API returned an error: 0, message='Attempt to decode JSON with unexpected mimetype: text/html', url=URL('https://www.soliscloud.com:13333/v1/api/stationYear'), error code: None, response None
> station_all(station_id = xxxxxxxxx ): Failed with: API returned an error: 0, message='Attempt to decode JSON with unexpected mimetype: text/html', url=URL('https://www.soliscloud.com:13333/v1/api/stationAll'), error code: None, response None
> Exceptions:  [None]

_

**Comment by heli051** [comment]
Thanks to those who suggested removing the trailing slash. It worked a treat but no data yet. I shall be patient...
